server/api/availability.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package api
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/robfig/cron/v3"
+
+	"github.com/mattermost/mattermost-plugin-msoffice/server/job"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/remote"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/utils"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/utils/bot"
+)
+
+const (
+	AVAILABILITY_VIEW_FREE              = '0'
+	AVAILABILITY_VIEW_TENTATIVE         = '1'
+	AVAILABILITY_VIEW_BUSY              = '2'
+	AVAILABILITY_VIEW_OUT_OF_OFFICE     = '3'
+	AVAILABILITY_VIEW_WORKING_ELSEWHERE = '4'
+)
+
+type availabilityJob struct {
+	api API
+}
+
+func NewAvailabilityJob(api API) job.RecurringJob {
+	return &availabilityJob{api: api}
+}
+
+func (j *availabilityJob) Run() {
+	c := cron.New()
+	c.AddFunc("* * * * *", j.Work)
+	c.Start()
+}
+
+func (j *availabilityJob) getLogger() bot.Logger {
+	return j.api.(*api).Logger
+}
+
+func (j *availabilityJob) Work() {
+	log := j.getLogger()
+	log.Debugf("Availability job beginning")
+
+	_, err := j.api.GetUserAvailability()
+	if err != nil {
+		log.Errorf("Error during Availability job", "error", err.Error())
+	}
+
+	log.Debugf("Availability job finished")
+}
+
+func (api *api) GetUserAvailability() (string, error) {
+	client, err := api.MakeClient()
+	if err != nil {
+		return "", err
+	}
+
+	u, err := api.UserStore.LoadUser(api.mattermostUserID)
+	if err != nil {
+		return "", err
+	}
+
+	scheduleIDs := []string{u.Remote.Mail}
+
+	start, end, timeWindow := getTimeInfoForAvailability()
+
+	sched, err := client.GetSchedule(u.Remote.ID, scheduleIDs, start, end, timeWindow)
+	if err != nil {
+		return "", err
+	}
+
+	if len(sched) == 0 {
+		return "No schedule info found", nil
+	}
+
+	s := sched[0]
+	av := s.AvailabilityView[0]
+	return api.setUserStatusFromAvailability(api.mattermostUserID, av), nil
+}
+
+func (api *api) GetAllUsersAvailability() (string, error) {
```

Comment:
This is called by the recurring job. For debugging in development, you can trigger it with `/availability all`